pkg/routes/logs.go

```diff
@@ -42,10 +44,28 @@ func (l Logs) Install(c *restful.Container) {
 func logFileHandler(req *restful.Request, resp *restful.Response) {
 	logdir := "/var/log"
 	actual := path.Join(logdir, req.PathParameter("logpath"))
+
+	// check filename length first, return 404 if it's oversize.
+	if logFileNameIsTooLong(actual) {
+		http.Error(resp, "file not found", http.StatusNotFound)
+		return
+	}
 	http.ServeFile(resp.ResponseWriter, req.Request, actual)
 }
 
 func logFileListHandler(req *restful.Request, resp *restful.Response) {
 	logdir := "/var/log"
 	http.ServeFile(resp.ResponseWriter, req.Request, logdir)
 }
+
+// logFileNameIsTooLong checks filename length, returns true if it's longer than 255.
+// cause http.ServeFile returns default error code 500 except for NotExist and Forbidden, but we need to separate the real 500 from oversize filename here.
+func logFileNameIsTooLong(filePath string) bool {
+	_, err := os.Stat(filePath)
```

Comment:
does this work for platform other than linux?

Reply:
I think so?

https://golang.org/src/os/:

```
stat.go |   | 733
-- | -- | --
stat_aix.go |   | 1206
stat_darwin.go |   | 1216
stat_dragonfly.go |   | 1175
stat_freebsd.go |   | 1199
stat_js.go |   | 1250
stat_linux.go |   | 1189
stat_netbsd.go |   | 1192
stat_openbsd.go |   | 1182
stat_plan9.go |   | 2353
stat_solaris.go |   | 1175
stat_test.go |   | 6121
stat_unix.go |   | 1404
stat_windows.go |   | 3641
```

I don't know how many of the OS implementations actually return this error though.